### PR TITLE
Implement the osu! logo visualizer

### PR DESCRIPTION
**Pull request by ColdVolcano**
Prereqs:
- [x] https://github.com/ppy/osu-framework/pull/841

### CODE REVIEW

File: osu.Game/Screens/Menu/LogoVisualisation.cs [hunk 1]
```diff
@@ -0,0 +1,198 @@
...
+{
+    internal class LogoVisualisation : Drawable, IHasAccentColour
+    {
+        private Bindable<WorkingBeatmap> beatmap = new Bindable<WorkingBeatmap>();
+        private Color4 barColour;
+
+        private const int index_change = 5;
+        private const int bar_length = 1200;
+        private const int bars_per_visualizer = 250;
+        private const int visualizers = 5;
```
Comment (smoogipooo) [inline osu.Game/Screens/Menu/LogoVisualisation.cs]: Can this be a float for now? Keep it as a const but we may want to tween it later on. Along with bar_length.

**Review by smoogipoo — DISMISSED** [review]
Good stuff!

File: osu.Game/Screens/Menu/LogoVisualisation.cs [hunk 2]
```diff
@@ -0,0 +1,198 @@
...
+namespace osu.Game.Screens.Menu
+{
+    internal class LogoVisualisation : Drawable, IHasAccentColour
+    {
+        private Bindable<WorkingBeatmap> beatmap = new Bindable<WorkingBeatmap>();
+        private Color4 barColour;
+
+        private const int index_change = 5;
+        private const int bar_length = 1200;
+        private const int bars_per_visualizer = 250;
```
Comment (ColdVolcano) [inline osu.Game/Screens/Menu/LogoVisualisation.cs]: to keep all the 5 visualizers in the same rotation.
256 would look like this: 
image

Comment (Nabile-Rahmani) [inline osu.Game/Screens/Menu/LogoVisualisation.cs]: Last time I counted the stable visualiser, there were 4 layers of 256 bars IIRC.
Gotta love powers of two.

Comment (ColdVolcano) [inline osu.Game/Screens/Menu/LogoVisualisation.cs]: 🤔 I counted 50 bars between big bars (bass) and they were 5 big bars, so I used that

**Comment by peppy** [comment]
Nice work on this.

**Review by peppy — APPROVED** [review]

**Pull request merged**